packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/PredefinedMetadataModal.tsx

```diff
@@ -0,0 +1,196 @@
+    options: [
+      { name: "Report", type: "text" },
+      { name: "History", type: "text" },
+      { name: "Changelog", type: "text" },
+    ],
+  },
+  {
+    name: "Boosted Pools",
+    options: [
+      { name: "Summary", type: "text" },
+      { name: "Changelog", type: "text" },
+      { name: "Risks", type: "text" },
+    ],
+  },
+];
+
+interface ICheckboxState {
+  [key: string]: {
+    [key: string]: boolean;
+  };
+}
+
+interface PoolMetadataAttribute {
+  typename: (typeof TypenameEnum.enum)[keyof typeof TypenameEnum.enum];
+  key: string;
+  description: null;
+  value: null;
+}
+
+export function PredefinedMetadataModal({ close }: { close?: () => void }) {
+  const [checkboxes, setCheckboxes] = useState<ICheckboxState>(
+    ATTRIBUTES_TEMPLATE.reduce((state, attribute) => {
+      const options = attribute.options.reduce(
+        (opt, option) => ({
+          ...opt,
+          [option.name]: false,
+        }),
+        { selectAll: false }
+      );
+      return { ...state, [attribute.name]: options };
+    }, {} as ICheckboxState)
+  );
+
+  const { handleAddMetadata, isKeyUnique } = usePoolMetadata();
+
+  const handleCheckboxChange = (groupName: string, optionName: string) => {
+    setCheckboxes((prevCheckboxes) => {
+      const updatedCheckboxes = {
+        ...prevCheckboxes,
+        [groupName]: {
+          ...prevCheckboxes[groupName],
+          [optionName]: !prevCheckboxes[groupName][optionName],
+        },
+      };
+      if (optionName === "selectAll") {
+        Object.keys(updatedCheckboxes[groupName]).forEach((key) => {
+          updatedCheckboxes[groupName][key] =
+            updatedCheckboxes[groupName]["selectAll"];
+        });
+      } else if (!updatedCheckboxes[groupName][optionName]) {
+        updatedCheckboxes[groupName].selectAll = false;
+      }
+      return updatedCheckboxes;
+    });
+  };
+
+  const getSelectedAttributesAndTypes = (): PoolMetadataAttribute[] => {
+    const selectedAttributes: PoolMetadataAttribute[] = [];
+    for (const attribute of ATTRIBUTES_TEMPLATE) {
+      for (const option of attribute.options) {
+        const { name, type } = option;
+        if (checkboxes[attribute.name][name]) {
+          selectedAttributes.push({
+            typename: type,
+            key: toSlug(name),
+            value: null,
+            description: null,
+          });
+        }
+      }
+    }
+    return selectedAttributes;
+  };
+
+  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const selectedAttributes = getSelectedAttributesAndTypes();
+    selectedAttributes.map((item) => {
+      if (isKeyUnique(item.key)) {
+        handleAddMetadata(item);
+      }
+    });
+    close?.();
+  };
+
+  return (
+    <form onSubmit={onSubmit}>
+      <Accordion.Root
+        className="my-4 w-full"
+        type="single"
+        defaultValue="item-1"
+        collapsible
+      >
+        {ATTRIBUTES_TEMPLATE.map((attribute) => (
```

Comment:
`#tip` you can destructure elements here to make your code easier to parse
```suggestion
        {ATTRIBUTES_TEMPLATE.map(({ name, options }) => (
```